src/components/FieldValueDirectAnswer.tsx

```diff
@@ -0,0 +1,124 @@
+
+  return (
+    <div className={cssClasses.answerContainer}>
+      {title &&
+      <div className={cssClasses.header}>{title}</div>}
+      <div className={cssClasses.content}>
+        <div className={cssClasses.body}>{resultContent}</div>
+        {link && <div className='mt-4'>
+          <a href={link} className='text-primary' onClick={viewDetailsClickHandler}>
+            View Details
+          </a>
+        </div>
+        }
+      </div>
+    </div>
+  );
+}
+
+function getResultContent(result: FieldValueDirectAnswerType): JSX.Element {
+  switch (result.fieldType) {
+    case BuiltInFieldType.InstagramHandle:
+      return getUrlJsxElement(`https://www.instagram.com/${result.value}`, result.value);
+    case BuiltInFieldType.TwitterHandle:
+      return getUrlJsxElement(`https://twitter.com/${result.value}`, `@${result.value}`);
+    case BuiltInFieldType.FacebookURL:
+    case BuiltInFieldType.AndroidAppURL:
+    case BuiltInFieldType.IOSAppURL:
+      return getUrlJsxElement(result.value);
+    case BuiltInFieldType.ComplexURL:
+      const url = result.value.url;
+      const displayUrl = result.value.preferDisplayUrl ? result.value.displayUrl : url;
+      return getUrlJsxElement(url, displayUrl);
+    case BuiltInFieldType.URL:
+      return Array.isArray(result.value)
+        ? getListJsxElement(result.value, url => getUrlJsxElement(url))
+        : getUrlJsxElement(result.value);
+    case BuiltInFieldType.Phone:
+      return getUrlJsxElement(`tel:${result.value}`, result.value);
+    case BuiltInFieldType.Email:
+      return getListJsxElement(result.value, e => getUrlJsxElement(`mailto:${e}`, e));
+    case BuiltInFieldType.Address:
+      return getAddressJsxElement(result.value);
+    case BuiltInFieldType.RichText:
+      return <div>{result.value}</div>; //TODO: SLAP-2340
+    case 'unknown':
+      return <div>unknown</div>; //TODO: SLAP-2337
+    default:
+      return Array.isArray(result.value)
+        ? getListJsxElement(result.value, val => <p className='whitespace-pre-wrap'>{val}</p>)
+        : <p className='whitespace-pre-wrap'>{result.value}</p>;
+  }
+}
+
+function getListJsxElement<T>(
+  list: T[],
+  getItemJsxElement: (el: T) => JSX.Element
+): JSX.Element {
+  return (<ul className='list-disc list-inside'>
+    {list.map((el, i) =>
+      <li key={i}>
+        {getItemJsxElement(el)}
+      </li>)}
+  </ul>);
+}
+
+function getUrlJsxElement(href: string, displayText?: string): JSX.Element {
+  return <a href={href} className='text-primary'>{displayText ?? href}</a>;
+}
+
+function getAddressJsxElement(address: Address): JSX.Element {
+  // user specified display Address in KM
+  if (address.extraDescription) {
+    return <div>{address.extraDescription}</div>;
+  }
+  return <div>
+    {address.line1 && <p>{address.line1}</p>}
+    {address.line2 && <p>{address.line2}</p>}
+    {address.line3 && <p>{address.line3}</p>}
+    <p>{address.city}, {address.region} {address.postalCode}</p>
```

Comment:
are these fields all required?

Reply:
I thought only `countryCode` was required, so maybe we want to check that `city`, `region`, and `postalCode` exist before displaying them? otherwise it might look weird if, for example, there was no `city`

Reply:
since `city`, `region`, and `postalCode` are only required after a `countryCode` is set, I was thinking maybe we'd check for those after checking for `countryCode`?

Reply:
updated to optionally display city, region, and postalCode